app/3.0/toolbox/xss.md

```diff
@@ -0,0 +1,218 @@
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: block;
+        }
+      </style>
+      <h2>Random: ${Math.random()}</h2>
+    `;
+  }
+}
+```
+
+### Implement Polymer’s sanitizeDOMValue callback {#sanitizedomvalue}
+
+An [XSS sink](https://github.com/wisec/domxsswiki/wiki/Sinks) is a location in which an attacker could potentially place malicious code, with the intention that your app or element will unintentionally execute that code.
+
+Properties and attributes that are only ever treated as text are not vulnerable to these kinds of attacks. However, an attacker could potentially exploit element properties and attributes like `src`, `href`, and `onclick` as XSS sinks.
+
+To safeguard against this type of attack, implement Polymer's `sanitizeDOMValue` callback.
+
+`sanitizeDOMValue` is a global callback that Polymer runs on all values directly before they are processed in a property or attribute binding, or synchronized to an attribute with `reflectToAttribute`.
+
+All versions of Polymer include the `sanitizeDOMValue` hook.
+
+Polymer does not ship with a default implementation of `sanitizeDOMValue`. However, a useful implementation might inspect the property or attribute being set, and either reject, sanitize, or require trusted value types before setting the value. 
+
+Set your sanitizer function from your app entrypoint (usually index.html), before loading your app shell or any Polymer elements:
+
+index.html {.caption}
+
+```html
+<script type="module">
+  // import the setSanitizeDOMValue method from Polymer settings utils
+  import { setSanitizeDOMValue } from '/node_modules/@polymer/polymer/lib/utils/settings.js';
+
+  // call setSanitizeDOMValue on the function that performs sanitization
+  setSanitizeDOMValue((value, name, type, node) => {
+    // sanitze value here
+    return sanitizedValue;
+  });
+</script>
+
+<!-- now it's safe to load element definitions and upgrade elements -->
+<script type="module" src="my-app.js"></script>
+```
+
+See the [API docs on `sanitizeDOMValue`](/{{{polymer_version_dir}}}/docs/devguide/settings) for more information.
+
+## Implement standard web security practices {#standardsecurity}
+
+### Avoid using innerHTML to set untrusted values {#avoidinnerhtml}
+
+Developers using any web technology must take extreme care with APIs like [innerHTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML) and [insertAdjacentHTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML) that directly manipulate DOM. If you use these APIs, you must guarantee that any HTML strings you pass to them are free from possibly malicious code. See the [MDN documentation on innerHTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML) for more info.
+
+Polymer developers , an extra consideration applies to these APIs, in that we must also rule out the evaluation of untrusted custom element templates.
```

Comment:
Eliminate "we" -- suggest maybe "Polymer developers must also guard against the evaluation of untrusted code in custom element templates."